### DISCUSSION TITLE
Spectrogram issue

### DISCUSSION THREAD

**Discussion by sjg2203** [raphaelvallat/yasa#101] [Q&A] [answered]
When trying to generate spectrograms, I always end up with a 15h plot instead of an 8h one (see the plot attached).

```python
input_fname=easygui.fileopenbox(msg="EDF file")
raw=mne.io.read_raw_edf(input_fname,preload=True,verbose=False)
print(raw.info['sfreq'])
raw.resample(100)
sf=raw.info['sfreq']
print(sf)
raw.filter(0.1,45)
data=raw.get_data(units="uV")
#Transform tuple to np.array to get a 1D array
from typing import Iterable
C3=raw['C3']
def flatten(items):
  """Yield items from any nested iterable."""
   for x in items:
      if isinstance(x,Iterable) and not isinstance(x,(str,bytes)):
      for sub_x in flatten(x):
          yield sub_x
      else:
          yield x
C3=list(flatten(C3))
C3=np.array(C3)
#Spectrogram plot
yasa.plot_spectrogram(C3,sf);
plt.show()
```
![Spectrogram](https://user-images.githubusercontent.com/58807831/200350642-a83243ae-b697-4cfa-8caa-95c7ecbbe0b5.png)

I also tried to select my electrode by using `C3=raw.pick_channels(['C3'])` instead of `C3=raw['C3']` and then transforming it to fit a 1D array, but it didn't work either.

If anyone can help elucidate this bit of an issue, that'd be much appreciated.
Thanks

**Comment by remrama** [comment, marked as answer]
I'm not sure what's happening with the extra steps to flatten the data, and skeptical something there might be causing this. Have you looked at the length of the final array you have before plotting it to make sure it's as long as you think it should be? Can you try using just this to extract the one-dimensional array of C3 data and plot it?
```python
C3 = raw.get_data(picks=["C3"], units="uV")[0]
yasa.plot_spectrogram(C3, sf)
plt.show()
```

If that doesn't work, could you help with some more troubleshooting by showing the length of the file? Maybe the output from `print(raw.times[0], raw.times[-1])`?

**Reply by sjg2203** [reply]
Thanks @remrama, it solved my issue!


For your other inquiry, here is what resulted:
```python
print(raw.times[0],raw.times[-1])
0.0 25695.99
```
